### ISSUE TITLE
Questions about token-exipre

### ISSUE THREAD

**Issue by AlexZeDim** [QuadDamn/battlenet-api-wrapper#18]
Hello there, hope this better then [Blizzard.js](https://github.com/benweier/blizzard.js)

So, I have seen the source code of the project [battleNetWrapper.ts](https://github.com/QuadDamn/battlenet-api-wrapper/blob/master/src/battleNetWrapper.ts) and have a question about it's functionality. 

As I understand it, in such code as: 

```
//we take every `${name}` from database

(async function() {
    const bnw = new battleNetWrapper();
    await bnw.init(BNET_ID, BNET_SECRET, 'eu', 'en_GB'); 
    for (let name = await cursor.next(); name != null; name = await cursor.next()) {
        const data = await bnw.WowProfileData.getCharacterEquipment(realm, name);
        console.log(data);
    }
}());
```

actual token stores right at this stage:
    `await bnw.init(BNET_ID, BNET_SECRET, 'eu', 'en_GB'); `

so if I want to update the entire DB of players one-by-one, which have millions of names, I guess the `token` will expire during the execution of this stage, cause it takes a lot of time for updating every character in DB.

So, the question is not: 

> How to prevent it and keep token up-to-date?

but:

> How to manually insert/refresh token during the execution of `for loop` stage with using of `battlenet-api-wrapper`

Does this module support this kind of functional? Can I manually insert my token during `init` stage?

**Comment by QuadDamn** [comment]
@AlexZeDim I'll dig into it tomorrow and get you an answer.  It may require some code modification on my end!

**Comment by QuadDamn** [comment]
@AlexZeDim Check out https://github.com/QuadDamn/battlenet-api-wrapper/pull/21

You should be able to call the `setOAuthToken` function from the instantiated class and it will refresh the token and assign it to the `oauthToken` class variable for subsequent calls.

I just tested it myself and if you make the calls too quickly together (tried a minute wait between the assignments) you will get the same oauth token.  So, they must have a memoizee-type cache on their oauth token endpoint or something.

Please let me know if you think this will do the trick for you and I will get it merged in and get a new npm package version out.

**Comment by AlexZeDim** [comment]
Yes, thank you. This is the exactly what I want.

**Comment by QuadDamn** [comment]
@AlexZeDim Went ahead and merged in #21.  I released a new package version to the NPM registry.  You should be good to update to the latest version and see if that fixes your problem.  Let me know if it doesn't.
